modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIStoreRestClient.java

```diff
@@ -1261,4 +1261,35 @@ public HttpResponse changePassword(String username, String currentPassword, Stri
             throw new APIManagerIntegrationTestException("Unable to change password. Error: " + e.getMessage(), e);
         }
     }
+
+    /**
+     * Add application with custom attributes
+     *
+     * @param application - application  name
+     * @param tier        - throttling tier
+     * @param callbackUrl - callback url
+     * @param description - description of app
+     * @param applicationAttributes - Json string of custom attributes defined by user
+     * @return - http response of add application
+     * @throws org.wso2.am.integration.test.utils.APIManagerIntegrationTestException - if fails to add application
+     */
+    public HttpResponse addApplicationWithCustomAttributes (String application, String tier, String callbackUrl,
+                                       String description, String applicationAttributes)
+            throws APIManagerIntegrationTestException {
+        try {
+            checkAuthentication();
+            String urlAppAttributes = URLEncoder.encode(applicationAttributes, "UTF-8");
+            return HTTPSClientUtils.doPost(
+                    new URL(backendURL +
+                            "store/site/blocks/application/application-add" +
+                            "/ajax/application-add.jag?action=addApplication&tier=" +
+                            tier + "&callbackUrl=" + callbackUrl + "&description=" + description +
+                            "&application=" + application + "&applicationAttributes=" +
+                            urlAppAttributes), "", requestHeaders);
+        } catch (Exception e) {
```

Comment:
Instead of generic exception, can we catch the relevant exception(s) here. I think it is the best practice :)

Reply:
you can log the error message here as well.
string message = ""Unable to add application - " + application  + " with custom attributes. Error: " + e.getMessage()";
log.error(message);

Reply:
Resolved.